ddl/ddl.go

```diff
@@ -1462,9 +1462,11 @@ func cancelRunningJob(sess *sess.Session, job *model.Job,
 // pauseRunningJob check and pause the running Job
 func pauseRunningJob(sess *sess.Session, job *model.Job,
 	byWho model.AdminCommandOperator) (err error) {
-	// It would be much better doing this filter during `getJobsBySQL`, but not now.
+	if job.IsPausing() || job.IsPaused() {
+		return dbterror.ErrPausedDDLJob.GenWithStackByArgs(job.ID)
+	}
 	if !job.IsPausable() {
-		err = dbterror.ErrCannotPauseDDLJob.GenWithStackByArgs(job.ID)
+		err = dbterror.ErrCannotPauseDDLJob.GenWithStackByArgs(job.ID, job.State.String(), job.SchemaState.String())
```

Comment:
The message should be updated when you put more arguments to the error.

Error message:
```
ErrCannotPauseDDLJob:  mysql.Message("Job [%v] can't be paused now: %s", nil),
```

When you call `GenWithStackByArgs`:
```
fmt.Sprintf("state: %s, schema state: %s", job.State.String(), job.SchemaState.String())
```

Reply:
Already done that.